json/schemas/partition-table-schema.json

```diff
@@ -0,0 +1,175 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "title": "Partition Table",
+    "description": "Layout of the partition table",
+    "type": "object",
+    "properties": {
+        "$schema": {},
+        "version": {
+            "description": "Partition Table Version",
+            "type": "array",
+            "prefixItems": [
+                {
+                    "description": "Major Version",
+                    "type": "integer",
+                    "minimum": 0
+                },
+                {
+                    "description": "Minor Version",
+                    "type": "integer",
+                    "minimum": 0
+                }
+            ]
+        },
+        "unpartitioned": {
+            "description": "Unpartitioned space UF2 families and permissions",
+            "type": "object",
+            "properties": {
+                "families": {
+                    "description": "UF2 families accepted",
+                    "type": "array",
+                    "items": {
+                        "enum": [
+                            "data",
+                            "absolute",
+                            "rp2040",
+                            "rp2350-arm-s",
+                            "rp2350-arm-ns",
+                            "rp2350-riscv"
+                        ]
+                    }
+                },
+                "permissions": {
+                    "description": "Permissions",
+                    "type": "object",
+                    "properties": {
+                        "secure": {
+                            "description": "Secure Permissions",
+                            "type": "string",
+                            "pattern": "^(r|w){0,2}$"
+                        },
+                        "nonsecure": {
+                            "description": "Non-Secure Permissions",
+                            "type": "string",
+                            "pattern": "^(r|w){0,2}$"
+                        },
+                        "bootloader": {
+                            "description": "Bootloader Permissions",
+                            "type": "string",
+                            "pattern": "^(r|w){0,2}$"
+                        }
+                    }
+                }
+            },
+            "required": ["permissions", "families"],
```

Comment:
If `permissions` is required, are the sub-properties within `permissions` also required?

Reply:
No, just an empty permissions will work fine
```
"permissions": {}
```